src/main/java/at/hannibal2/skyhanni/data/maxwell/MaxwellAPI.kt

```diff
@@ -31,35 +35,55 @@ import java.util.regex.Pattern
 @SkyHanniModule
 object MaxwellAPI {
 
-    private val storage get() = ProfileStorageData.profileSpecific
+    private val storage get() = ProfileStorageData.profileSpecific?.maxwell
 
-    var currentPower: String?
-        get() = storage?.maxwell?.currentPower
-        set(value) {
-            storage?.maxwell?.currentPower = value ?: return
+    var currentPower: MaxwellPower
+        get() = storage?.currentPower?.let(::getPowerByInternalNameOrNull) ?: NO_POWER
+        private set(value) {
+            if (storage?.currentPower == value.internalName) return
+            storage?.currentPower = value.internalName
+            MaxwellUpdateEvent.Power(value).post()
         }
 
-    var magicalPower: Int?
-        get() = storage?.maxwell?.magicalPower
-        set(value) {
-            storage?.maxwell?.magicalPower = value ?: return
+    var magicalPower: Int
+        get() = storage?.magicalPower ?: 0
+        private set(value) {
+            if (storage?.magicalPower == value) return
+            storage?.magicalPower = value
+            MaxwellUpdateEvent.MagicalPower(value).post()
         }
 
-    var tunings: List<ThaumaturgyPowerTuning>?
-        get() = storage?.maxwell?.tunings
-        set(value) {
-            storage?.maxwell?.tunings = value ?: return
+    var tunings: List<MaxwellTunings>
+        get() = storage?.tunings.orEmpty()
+        private set(value) {
+            if (storage?.tunings == value) return
+            storage?.tunings = value
+            MaxwellUpdateEvent.Tuning(value).post()
         }
 
-    var favoritePowers: List<String>
-        get() = storage?.maxwell?.favoritePowers.orEmpty()
+    var favoritePowers: List<MaxwellPower>
+        get() = storage?.favoritePowers?.mapNotNull(::getPowerByInternalNameOrNull).orEmpty()
         set(value) {
-            storage?.maxwell?.favoritePowers = value
+            storage?.favoritePowers = value.map(MaxwellPower::internalName)
         }
 
-    private val NO_POWER by lazy { getPowerByNameOrNull("No Power") }
-    private var powers = mutableListOf<String>()
+    var inInventory: Boolean = false
+        private set
+
+    var inTuningGui: Boolean = false
+        private set
+
+    fun getPowerByNameOrNull(name: String): MaxwellPower? = powers.values.find { it.name == name }
 
+    fun getPowerByInternalNameOrNull(name: String): MaxwellPower? = powers[name]
+
+    val NO_POWER = MaxwellPower("No Power", "NO_POWER")
+
+    private var powers = mapOf<String, MaxwellPower>()
+    private const val THAUMATURGY_TUNINGS_SLOT = 51
+    private const val THAUMATURGY_MP_SLOT = 48
+
+    //region Patterns
```

Comment:
Not saying `//region` is explicitly wrong, but parts of the code base are already using `// <editor-fold>` & `// </editor-fold>`, which can be utilized with the current rules intact, and this way we wouldn't have to add extra edge-case detection to the CustomCommentSpacing rule.